### PR TITLE
Don't request rates if settings are empty.

### LINKED ISSUE #193: Do not run rate requests for instances of USPS with no saved settings
i.e. if you add WooCommerce Connect USPS to a Zone but never Save Changes on its settings page, then the settings for that instance are probably empty in options.

We should 1) not generate shipping rate requests from instances in that state and 2) log when a shipping rate request was so requested.

Otherwise, we will probably log confusing messages as rate requests fail (since things like origin zip code which is required will be empty.)

### PR DESCRIPTION
**Pull request by jeffstieler**
Fixes #193.

To test:
* Add a new USPS shipping instance to a zone and do not enter any settings for it
* Add an item to cart and attempt a checkout (that would fall under the above zone)
* Verify that a "Service settings empty." warning is logged

### CODE REVIEW

File: tests/php/test_class-wc-connect-shipping-method.php [hunk 1]
```diff
@@ -99,4 +99,64 @@ public function test_is_valid_package_destination( $package, $expected ) {
...
+			->will( $this->returnValue( false ) );
+
+		$shipping_method->expects( $this->once() )
+			->method( 'log' )
+			->with(
+				$this->stringContains( 'Package destination failed validation.' ),
+				$this->anything()
+			);
+
+		$shipping_method->calculate_shipping();
```
Comment (jkudish) [inline tests/php/test_class-wc-connect-shipping-method.php]: should this test that `$shipping_method->calculate_shipping();` returns early?

File: tests/php/test_class-wc-connect-shipping-method.php [hunk 2]
```diff
@@ -99,4 +99,64 @@ public function test_is_valid_package_destination( $package, $expected ) {
...
+			->with(
+				$this->stringContains( 'Service settings empty.' ),
+				$this->anything()
+			);
+
+		$shipping_method->calculate_shipping();
+
+	}
+
 }
```
Comment (jkudish) [inline tests/php/test_class-wc-connect-shipping-method.php]: should we add a test for valid settings too?

**Comment by jkudish** [comment]
Added some feedback about tests, but otherwise this works well, tests well and code looks good.

File: tests/php/test_class-wc-connect-shipping-method.php (hunk 2 above)
Comment (jeffstieler) [inline tests/php/test_class-wc-connect-shipping-method.php]: I was trying to test the code I added without going overboard and getting full coverage for unrelated changes. Though yes, eventually this should be 100% covered.

File: tests/php/test_class-wc-connect-shipping-method.php (hunk 1 above)
Comment (jeffstieler) [inline tests/php/test_class-wc-connect-shipping-method.php]: In order to do so, we'd need to mock the API client as well, and so on. It's rather involved because we can't simply check the return value.. we have to make sure other methods weren't called.

Comment (jkudish) [inline tests/php/test_class-wc-connect-shipping-method.php]: For a future PR then perhaps

**Comment by jkudish** [comment]
:shipit:
